wrappers/services.go

```diff
@@ -614,44 +616,69 @@ func (es *ensureSnapServicesContext) ensureSnapSystemdUnits(snapInfo *snap.Info,
 		return nil
 	}
 
+	// lets sort the service list before generating them for
+	// consistency when testing
+	services := snapInfo.Services()
+	sort.Slice(services, func(i, j int) bool {
+		return services[i].Name < services[j].Name
+	})
+
 	// note that the Preseeding option is not used here at all
-	for _, app := range snapInfo.Apps {
-		if !app.IsService() {
+	for _, svc := range snapInfo.Services() {
+		// if an inclusion list is provided, then we want to make sure this service
+		// is included
+		snapServiceName := fmt.Sprintf("%s.%s", snapInfo.InstanceName(), svc.Name)
+		if len(es.opts.IncludeServices) > 0 && !strutil.ListContains(es.opts.IncludeServices, snapServiceName) {
 			continue
 		}
 
 		// create services first; this doesn't trigger systemd
 
-		// Generate new service file state
-		path := app.ServiceFile()
-		content, err := generateSnapServiceFile(app, opts)
+		// get the correct quota group for the service we are generating.
+		quotaGrp := opts.QuotaGroup
+		if quotaGrp != nil {
+			quotaGrp = quotaGrp.GroupForService(snapServiceName)
+			if quotaGrp == nil {
+				// default to the parent quota group, which may also be nil.
+				quotaGrp = opts.QuotaGroup
+			}
+		}
```

Comment:
nitpick: I still wonder if generate a map before the loop would be a bit better in terms of not going over the subgroups of QuotaGroup again and again

Reply:
Done! :-)